samza-kafka/src/main/scala/org/apache/samza/system/kafka/KafkaSystemFactory.scala

```diff
@@ -46,21 +43,25 @@ object KafkaSystemFactory extends Logging {
 class KafkaSystemFactory extends SystemFactory with Logging {
 
   def getConsumer(systemName: String, config: Config, registry: MetricsRegistry): SystemConsumer = {
-    val clientId = KafkaConsumerConfig.getConsumerClientId( config)
+    val clientIdPrefix = KafkaConsumerConfig.CONSUMER_CLIENT_ID_PREFIX;
     val metrics = new KafkaSystemConsumerMetrics(systemName, registry)
 
-    val kafkaConsumer = KafkaSystemConsumer.getKafkaConsumerImpl(systemName, clientId, config)
-    info("Created kafka consumer for system %s, clientId %s: %s" format (systemName, clientId, kafkaConsumer))
+    val kafkaConsumerConfig = KafkaConsumerConfig.getKafkaSystemConsumerConfig(config, systemName, clientIdPrefix);
```

Comment:
We pass in a constant  `KafkaConsumerConfig.CONSUMER_CLIENT_ID_PREFIX` defined in `KafkaConsumerConfig` to `KafkaConsumerConfig.getKafkaSystemConsumer` as `idPrefix`.  Why can't we inline the constant within the class itself?  Is there any scenarios that would need this?

Reply:
The idea is to provide ability to have a "custom" naming for the consumer, and not totally dependent on config.